util/mathutil/exponential_average.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mathutil
+
+// ExponentialMovingAverage is an exponential moving average measurement implementation. it is not thread-safe.
+type ExponentialMovingAverage struct {
+	value        float64
+	sum          float64
+	factor       float64
+	warmupWindow int
+	count        int
+}
+
+// NewExponentialMovingAverage will create a new ExponentialMovingAverage
+func NewExponentialMovingAverage(
+	factor float64,
+	warmupWindow int,
+) *ExponentialMovingAverage {
+	if factor >= 1 || factor <= 0 {
+		panic("factor must be (0, 1)")
```

Comment:
Could we just make it valid，instead of panic?

Reply:
of course，we can give it error to report error. but if developers doesn't check it. it will have problem. and we think it must success to create it so that we panic the program to avoid the illegal init.